src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.Changed.cs

```diff
@@ -93,4 +93,69 @@ public static void FileSystemWatcher_ChangeWatchedFolder()
             Utility.ExpectEvent(eventOccured, "changed");
         }
     }
+
+    [Fact]
+    public static void FileSystemWatcher_Changed_NestedDirectories()
+    {
+        using (var dir = Utility.CreateTestDirectory())
+        using (var watcher = new FileSystemWatcher())
+        {
+            AutoResetEvent eventOccured = Utility.WatchForEvents(watcher, WatcherChangeTypes.Changed);
+            watcher.Path = Path.GetFullPath(dir.Path);
+            watcher.Filter = "*";
+            watcher.IncludeSubdirectories = true;
+            watcher.EnableRaisingEvents = true;
+
+            using (var firstDir = new TemporaryTestDirectory(Path.Combine(dir.Path, "dir1")))
+            {
+                // We need to sleep to allow Linux FSW to get the event and get a WatchDescriptor on the directory
+                System.Threading.Thread.Sleep(1000);
```

Comment:
The point of these waits is to *not* proceed immediately on create due to a limitation of the Linux inotify API; once notified about a new create, the listening code (the FileSystemWatcher) must execute code to get a Watch Descriptor on that new directory (and traverse it if necessary) as well, otherwise we will not get notifications about changes underneath. The delay was necessary otherwise the tests fail on Linux since the FSW doesn't have time to create a WatchDescriptor and call inotify_add_watch before the new directory is created, thus missing changes.

I completely agree about having Sleep in test code, I'm just not sure what other options we have since that's exactly what we need to do in order to work around the limitations in the OS API...thoughts?